File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {

        // *** iterative BFS 

        if (root == null){
            return root;
        }

        Queue<TreeNode> lvl = new LinkedList<>();

        lvl.add(root);

        while (!lvl.isEmpty()){
            for (int i = 0; i < lvl.size(); i++){
                TreeNode c = lvl.remove();
                TreeNode l = c.left;
                TreeNode r = c.right;
                c.left = r;
                c.right = l;
                if (c.left != null){
                    lvl.add(c.left);
                }
                if (c.right != null){
                    lvl.add(c.right);
                }
            }
        }

        return root;

        
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        max_p = 0
        def max_depth(n):
            if not n:
                return 0

            return 1 + max(max_depth(n.left), max_depth(n.right))

        def max_depths(n):
            nonlocal max_p
            if not n:
                return
            left, right = max_depth(n.left), max_depth(n.right)
            max_p = max(left + right, max_p)

            max_depths(n.left)
            max_depths(n.right)

        max_depths(root)
        return max_p

        
